src/serializer/index.ts

```diff
@@ -43,19 +43,21 @@ export class Serializer {
 		});
 
 		const stream = await strategy.deserialize(base);
-		const result: Promise<T> = new Promise((resolve) => {
+		base.push(data);
+		base.push(null);
+		return new Promise<T>((resolve) => {
```

Comment:
I just changed the order because the promise callback is executed after the pushes either way